src/solver/conv_hip_implicit_gemm_3d_grouped_bwd_xdlops.cpp

```diff
@@ -341,28 +360,29 @@ ConvSolution ConvHipImplicitGemm3DGroupBwdXdlops::GetSolution(
     [[maybe_unused]] const PerformanceConfigHipImplicitGemm3DGroupBwdXdlops& config) const
 {
 #if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
-    switch(problem.GetInDataType())
-    {
-    case miopenInt8:
-        return MakeInvokerFactory<DeviceOpGBwdPtrs<int8_t>, CKArgs, conv::DataInvokeParams>(
-            problem, config.kernel_id);
-    case miopenHalf:
-        return MakeInvokerFactory<DeviceOpGBwdPtrs<ck::half_t>, CKArgs, conv::DataInvokeParams>(
-            problem, config.kernel_id);
-    case miopenFloat:
-        return MakeInvokerFactory<DeviceOpGBwdPtrs<float>, CKArgs, conv::DataInvokeParams>(
-            problem, config.kernel_id);
-    case miopenInt32:
-    case miopenBFloat16:
-    case miopenDouble:
-    case miopenFloat8:
-    case miopenBFloat8:
-    default:
-        MIOPEN_THROW(miopenStatusInternalError,
-                     "ConvHipImplicitGemmBwdXdlops operation not implemented for this data type");
-    }
-#endif
+    return MakeSolutionGroupConvImplicitGemmXdlops(
+        problem,
+        [&](auto data_type_val) {
+            using T = decltype(data_type_val);
+            /// \todo This call should be InitInvokerFactoryBwdNCHW but due to the
+            /// silliness of "in tensor is out and out is in" in backward pass,
+            /// InitInvokerFactoryFwdNCHW works correct while Bwd call causes wrong
+            /// output -- amberhassaan
+            return InitInvokerFactoryFwdNCHW<3,
```

Comment:
I agree with the confusion about the tensors in bwd. However, if you switched the input and output tensors, then the bwd should work properly.

Reply:
Calling the `InitInvokerFactoryFwd...` has the exact same effect.

Reply:
@iq136boy @amberhassaan as I recall there is a "unified 2d convolution problem" which automatically swaps all the tensors and dimensions for fwd\bwd\wrw passes. But I didn't did deeper and probably it doesn't solve all the problems.